client/client.go

```diff
@@ -413,7 +417,57 @@ func (client *Client) doSync(method, path string, query url.Values, headers map[
 	return client.doSyncWithOpts(method, path, query, headers, body, v, nil)
 }
 
+// checkMaintenanceJSON checks if there is a maintenance.json file written by
+// snapd the daemon that positively identifies snapd as being unavailable due to
+// maintenance, either for snapd restarting itself to update, or rebooting the
+// system to update the kernel or base snap, etc. If there is ongoing
+// maintenance, then the maintenance object on the client is set appropriately.
+// note that currently checkMaintenanceJSON does not return non-nil errors, so
+// if the file is missing or corrupt or empty, nothing will happen
+func (client *Client) checkMaintenanceJSON() error {
```

Comment:
AFAICT we never return an error in this function - should we just remove that return? YAGNI and all that (but of course only if we don't have short term plans where this actually would error).

Reply:
yes I think we could just not use an error return type here for now, it's easy enough to add one later, and for the most parts errors here should just be ignored I think anyways